pricegraph/src/api/price_estimation.rs

```diff
@@ -125,6 +129,10 @@ impl Pricegraph {
         while let Some(flow) = orderbook
             .fill_optimal_transitive_order_if(inverse_pair, |flow| flow.exchange_rate <= max_xrate)
         {
+            if flow.is_dust_trade() {
+                continue;
```

Comment:
Is it my impression that anyplace where a "continue" is needed, the logic could be expressed in a different way. Like filtering out the dust trades before entering the loop or even, 
including the filter here:
```
fill_optimal_transitive_order_if(inverse_pair, |flow| !flow.is_dust_trade() && flow.exchange_rate <= max_xrate)
```

Of course, this is fine as is.

Reply:
The issue is if we do that, then we will `break` when we encounter a dust order instead of skipping it. Alternatively, one cool way we could solve this is another helper method returning an `Iterator` over transitive orders and then use the `filter` method.